src/components/TodoItem.tsx

```diff
@@ -0,0 +1,119 @@
+import { useEffect, useRef, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  todo: Todo,
+  isActive?: boolean,
+  isProcessing: boolean,
+  handleDeleteTodo: (id: number) => void,
+  handleStatusUpdate: (todo: Todo) => void,
+  handleTitleUpdate: (todo: Todo, newTitile: string) => Promise<void>,
+};
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  handleDeleteTodo,
+  handleStatusUpdate,
+  handleTitleUpdate,
+  isProcessing,
+}) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [newTitle, setNewTitle] = useState(todo.title);
+
+  const inputUpdateRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (inputUpdateRef.current) {
+      inputUpdateRef.current.focus();
+    }
+  }, [isEditing]);
+
+  const handleTodoDoubleClick = () => {
+    setIsEditing(true);
+    setNewTitle(todo.title);
+  };
+
+  const handleFormSubmit = (event:
+  React.FocusEvent<HTMLInputElement>
+  | React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (newTitle === todo.title) {
+      return;
+    }
```

Comment:
```suggestion
    if (newTitle === todo.title) {
      setIsEditing(false);
      return;
    }
```
Maybe you should add `setIsEditing(false);` here, I can't close editing if the title is the same